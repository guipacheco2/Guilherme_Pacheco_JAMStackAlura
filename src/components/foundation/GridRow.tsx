import React from 'react'
import styled from 'styled-components'

const StyledGridRow = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-right: -16px;
  margin-left: -16px;
`

interface GridRowProps {
  children: React.ReactNode
}

export function GridRow({ children }: GridRowProps): JSX.Element {
  return <StyledGridRow>{children}</StyledGridRow>
}
