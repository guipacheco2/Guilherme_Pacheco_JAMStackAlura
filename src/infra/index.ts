export * from './hooks'
export * from './test'
