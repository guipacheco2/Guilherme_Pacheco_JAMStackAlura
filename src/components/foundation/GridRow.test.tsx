import React from 'react'
import { render } from '../../infra'
import { GridRow } from './GridRow'

describe('<GridRow />', () => {
  test('renders component', () => {
    const { baseElement } = render(<GridRow>child</GridRow>)
    expect(baseElement).toMatchSnapshot()
  })
})
