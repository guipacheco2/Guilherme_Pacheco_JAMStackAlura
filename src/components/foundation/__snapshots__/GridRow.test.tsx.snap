// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridRow /> renders component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -16px;
  margin-left: -16px;
  height: auto;
}

<body>
  <div>
    <div
      class="c0"
    >
      child
    </div>
  </div>
</body>
`;
